# Grouping and Aggregating Variables

1. Import Libraries and Dataframe
2. Aggregate Data\
    a. By Subset\
    b. By Complete Dataframe
3. Create Loyalty Flag\
    a. Create Loyalty Flag\
    b. Spending Habits Based on Loyalty\
    c. Investigate Max Price Outlier
4. Create Spending Flag\
    a. Create Activity Flag\
    b. Create Spending Flag Variable Based on Each Customer's Spending
5. Create Frequency Flag\
    a. Create Variable for Median Order Frequency\
    b. Create Frequency Flag from med_ord_freq
6. Export Data

## 1. Import Libraries and Dataframe

In [100]:
# import libraries
import pandas as pd
import numpy as np
import os

In [101]:
# data folder path
path_git = r'/Users/catherinemikelson/Desktop/Career Foundry/Acheivement 4 - Python Fundmentals/Instacart Basket Analysis Github'

In [102]:
# import ords_prods_merged_2
ords_prods_merged = pd.read_pickle(os.path.join(path_git, '02 Data', 'ords_prods_merged_2.pkl'))

In [103]:
ords_prods_merged.shape

(32404859, 19)

In [104]:
ords_prods_merged.head()

order_id  user_id  order_number  orders_day_of_the_week  order_hour_of_day  \
0   2539329        1             1                       2                  8   
1   2398795        1             2                       3                  7   
2    473747        1             3                       3                 12   
3   2254736        1             4                       4                  7   
4    431534        1             5                       4                 15   

   days_since_prior_order  new_customer  product_id  add_to_cart_order  \
0                     NaN          True         196                  1   
1                    15.0         False         196                  1   
2                    21.0         False         196                  1   
3                    29.0         False         196                  1   
4                    28.0         False         196                  1   

   reordered _merge product_name  aisle_id  department_id  prices  \
0          0   both         Soda        77              7     9.0   
1          1   both         Soda        77              7     9.0   
2          1   both         Soda        77              7     9.0   
3          1   both         Soda        77              7     9.0   
4          1   both         Soda        77              7     9.0   

     price_range_loc     busiest_day         busiest_days    busiest_hour  
0  Mid-range product  Regularly busy  Regularly busy days  average_orders  
1  Mid-range product  Regularly busy      Least busy days  average_orders  
2  Mid-range product  Regularly busy      Least busy days     most_orders  
3  Mid-range product      Least busy      Least busy days  average_orders  
4  Mid-range product      Least busy      Least busy days     most_orders

## 2. Aggregate Data

### a. By Subset

In [105]:
# create a subset of first 1,000,000 observations
df_subset = ords_prods_merged[:1000000]

In [106]:
# check shape
df_subset.shape

(1000000, 19)

In [107]:
# check output
df_subset.head(10)

order_id  user_id  order_number  orders_day_of_the_week  order_hour_of_day  \
0   2539329        1             1                       2                  8   
1   2398795        1             2                       3                  7   
2    473747        1             3                       3                 12   
3   2254736        1             4                       4                  7   
4    431534        1             5                       4                 15   
5   3367565        1             6                       2                  7   
6    550135        1             7                       1                  9   
7   3108588        1             8                       1                 14   
8   2295261        1             9                       1                 16   
9   2550362        1            10                       4                  8   

   days_since_prior_order  new_customer  product_id  add_to_cart_order  \
0                     NaN          True         196                  1   
1                    15.0         False         196                  1   
2                    21.0         False         196                  1   
3                    29.0         False         196                  1   
4                    28.0         False         196                  1   
5                    19.0         False         196                  1   
6                    20.0         False         196                  1   
7                    14.0         False         196                  2   
8                     0.0         False         196                  4   
9                    30.0         False         196                  1   

   reordered _merge product_name  aisle_id  department_id  prices  \
0          0   both         Soda        77              7     9.0   
1          1   both         Soda        77              7     9.0   
2          1   both         Soda        77              7     9.0   
3          1   both         Soda        77              7     9.0   
4          1   both         Soda        77              7     9.0   
5          1   both         Soda        77              7     9.0   
6          1   both         Soda        77              7     9.0   
7          1   both         Soda        77              7     9.0   
8          1   both         Soda        77              7     9.0   
9          1   both         Soda        77              7     9.0   

     price_range_loc     busiest_day         busiest_days    busiest_hour  
0  Mid-range product  Regularly busy  Regularly busy days  average_orders  
1  Mid-range product  Regularly busy      Least busy days  average_orders  
2  Mid-range product  Regularly busy      Least busy days     most_orders  
3  Mid-range product      Least busy      Least busy days  average_orders  
4  Mid-range product      Least busy      Least busy days     most_orders  
5  Mid-range product  Regularly busy  Regularly busy days  average_orders  
6  Mid-range product  Regularly busy         Busiest days     most_orders  
7  Mid-range product  Regularly busy         Busiest days     most_orders  
8  Mid-range product  Regularly busy         Busiest days     most_orders  
9  Mid-range product      Least busy      Least busy days  average_orders

In [108]:
# find order number mean, min, max
df_subset.groupby('department_id').agg({'order_number': ['mean', 'min', 'max']})

order_number        
                      mean min max
department_id                     
4                18.825780   1  99
7                17.472355   1  99
13               17.993423   1  99
14               19.246334   1  99
16               19.463012   1  99
17               11.294069   1  98
19               19.305237   1  99
20               17.599636   1  99

### b. By Complete Dataframe

In [109]:
# aggregate deptpartment_id order_number mean, min, max
ords_prods_merged.groupby('department_id').agg({'order_number': ['mean', 'min', 'max']})

order_number        
                      mean min max
department_id                     
1                15.457838   1  99
2                17.277920   1  99
3                17.170395   1  99
4                17.811403   1  99
5                15.215751   1  99
6                16.439806   1  99
7                17.225802   1  99
8                15.340650   1  99
9                15.895474   1  99
10               20.197148   1  99
11               16.170638   1  99
12               15.887671   1  99
13               16.583536   1  99
14               16.773669   1  99
15               16.165037   1  99
16               17.665606   1  99
17               15.694469   1  99
18               19.310397   1  99
19               17.177343   1  99
20               16.473447   1  99
21               22.902379   1  99

Observations: The result for the entire dataframe differs from the subset.  Firstly, with the subset, we had results for only 8 out of 20 departments.  The means change with the complete data frame.  For example, department 19's mean has changed from 19.3 to 17.2.  Department 17 changes from 11.3 to 15.7.

## 3. Create Loyalty Flag

### a. Create Loyalty Flag

In [110]:
# create max number - the number of orders by user_id
ords_prods_merged['max_order'] = ords_prods_merged.groupby(['user_id'])['order_number'].transform(np.max)

In [111]:
# check output
ords_prods_merged.head(10)

order_id  user_id  order_number  orders_day_of_the_week  order_hour_of_day  \
0   2539329        1             1                       2                  8   
1   2398795        1             2                       3                  7   
2    473747        1             3                       3                 12   
3   2254736        1             4                       4                  7   
4    431534        1             5                       4                 15   
5   3367565        1             6                       2                  7   
6    550135        1             7                       1                  9   
7   3108588        1             8                       1                 14   
8   2295261        1             9                       1                 16   
9   2550362        1            10                       4                  8   

   days_since_prior_order  new_customer  product_id  add_to_cart_order  \
0                     NaN          True         196                  1   
1                    15.0         False         196                  1   
2                    21.0         False         196                  1   
3                    29.0         False         196                  1   
4                    28.0         False         196                  1   
5                    19.0         False         196                  1   
6                    20.0         False         196                  1   
7                    14.0         False         196                  2   
8                     0.0         False         196                  4   
9                    30.0         False         196                  1   

   reordered _merge product_name  aisle_id  department_id  prices  \
0          0   both         Soda        77              7     9.0   
1          1   both         Soda        77              7     9.0   
2          1   both         Soda        77              7     9.0   
3          1   both         Soda        77              7     9.0   
4          1   both         Soda        77              7     9.0   
5          1   both         Soda        77              7     9.0   
6          1   both         Soda        77              7     9.0   
7          1   both         Soda        77              7     9.0   
8          1   both         Soda        77              7     9.0   
9          1   both         Soda        77              7     9.0   

     price_range_loc     busiest_day         busiest_days    busiest_hour  \
0  Mid-range product  Regularly busy  Regularly busy days  average_orders   
1  Mid-range product  Regularly busy      Least busy days  average_orders   
2  Mid-range product  Regularly busy      Least busy days     most_orders   
3  Mid-range product      Least busy      Least busy days  average_orders   
4  Mid-range product      Least busy      Least busy days     most_orders   
5  Mid-range product  Regularly busy  Regularly busy days  average_orders   
6  Mid-range product  Regularly busy         Busiest days     most_orders   
7  Mid-range product  Regularly busy         Busiest days     most_orders   
8  Mid-range product  Regularly busy         Busiest days     most_orders   
9  Mid-range product      Least busy      Least busy days  average_orders   

   max_order  
0         10  
1         10  
2         10  
3         10  
4         10  
5         10  
6         10  
7         10  
8         10  
9         10

In [112]:
#create loyalty flag
ords_prods_merged.loc[ords_prods_merged['max_order'] > 40, 'loyalty_flag'] = 'Loyal customer'
ords_prods_merged.loc[(ords_prods_merged['max_order'] <= 40) & (ords_prods_merged['max_order'] >10), 'loyalty_flag'] = 'Regular customer'
ords_prods_merged.loc[ords_prods_merged['max_order'] <= 10, 'loyalty_flag'] = 'New customer'

In [113]:
# check output
ords_prods_merged.head()

order_id  user_id  order_number  orders_day_of_the_week  order_hour_of_day  \
0   2539329        1             1                       2                  8   
1   2398795        1             2                       3                  7   
2    473747        1             3                       3                 12   
3   2254736        1             4                       4                  7   
4    431534        1             5                       4                 15   

   days_since_prior_order  new_customer  product_id  add_to_cart_order  \
0                     NaN          True         196                  1   
1                    15.0         False         196                  1   
2                    21.0         False         196                  1   
3                    29.0         False         196                  1   
4                    28.0         False         196                  1   

   reordered _merge product_name  aisle_id  department_id  prices  \
0          0   both         Soda        77              7     9.0   
1          1   both         Soda        77              7     9.0   
2          1   both         Soda        77              7     9.0   
3          1   both         Soda        77              7     9.0   
4          1   both         Soda        77              7     9.0   

     price_range_loc     busiest_day         busiest_days    busiest_hour  \
0  Mid-range product  Regularly busy  Regularly busy days  average_orders   
1  Mid-range product  Regularly busy      Least busy days  average_orders   
2  Mid-range product  Regularly busy      Least busy days     most_orders   
3  Mid-range product      Least busy      Least busy days  average_orders   
4  Mid-range product      Least busy      Least busy days     most_orders   

   max_order  loyalty_flag  
0         10  New customer  
1         10  New customer  
2         10  New customer  
3         10  New customer  
4         10  New customer

In [114]:
# check value counts
ords_prods_merged['loyalty_flag'].value_counts(dropna = False)

Regular customer    15876776
Loyal customer      10284093
New customer         6243990
Name: loyalty_flag, dtype: int64

### b. Spending Habits Based on Loyalty

Note: Is there a difference between the spending habits of customers based on Loyalty Status?

In [115]:
# basic statistical analysis of loyalty flag by prices
ords_prods_merged.groupby('loyalty_flag').agg({'prices': ['mean', 'min', 'max']})

prices              
                       mean  min      max
loyalty_flag                             
Loyal customer    10.386336  1.0  99999.0
New customer      13.294670  1.0  99999.0
Regular customer  12.495717  1.0  99999.0

In [116]:
# calculate loyalty percentage
ords_prods_merged['loyalty_flag'].value_counts(normalize = True)

Regular customer    0.489950
Loyal customer      0.317363
New customer        0.192687
Name: loyalty_flag, dtype: float64

Observations: Regular customers buy the highest priced items on average and make up the largest percentage of IC users (49%).
Additionlly, why is the max price 99,999?

### c. Investigate Max Price Outlier

In [117]:
# look for outliers
ords_prods_merged['prices'].drop_duplicates().sort_values(ascending = False)

29165516    99999.0
10030345    14900.0
18504754       25.0
25758883       24.9
28425432       24.8
             ...   
3384024         1.4
632766          1.3
5488887         1.2
5147325         1.1
60725           1.0
Name: prices, Length: 242, dtype: float64

In [118]:
# outliers
outliers = ords_prods_merged.loc[ords_prods_merged['prices'].isin([99999, 14900])]

In [119]:
# show all columns in dataset
pd.set_option('display.max_columns', None)

In [120]:
outliers['product_name'].value_counts(dropna = False)

Lowfat 2% Milkfat Cottage Cheese    4429
2 % Reduced Fat  Milk                698
Name: product_name, dtype: int64

Observation: If possible, the best next step would be to bring to check with the data source for the correct pricing of these two products. 

## 4. Create Spending Flag

### a. Create Activity Flag

In [121]:
# create average_price - the average price by user
ords_prods_merged['average_price'] = ords_prods_merged.groupby(['user_id'])['prices'].transform(np.mean)

In [122]:
# check output
ords_prods_merged.head(10)

order_id  user_id  order_number  orders_day_of_the_week  order_hour_of_day  \
0   2539329        1             1                       2                  8   
1   2398795        1             2                       3                  7   
2    473747        1             3                       3                 12   
3   2254736        1             4                       4                  7   
4    431534        1             5                       4                 15   
5   3367565        1             6                       2                  7   
6    550135        1             7                       1                  9   
7   3108588        1             8                       1                 14   
8   2295261        1             9                       1                 16   
9   2550362        1            10                       4                  8   

   days_since_prior_order  new_customer  product_id  add_to_cart_order  \
0                     NaN          True         196                  1   
1                    15.0         False         196                  1   
2                    21.0         False         196                  1   
3                    29.0         False         196                  1   
4                    28.0         False         196                  1   
5                    19.0         False         196                  1   
6                    20.0         False         196                  1   
7                    14.0         False         196                  2   
8                     0.0         False         196                  4   
9                    30.0         False         196                  1   

   reordered _merge product_name  aisle_id  department_id  prices  \
0          0   both         Soda        77              7     9.0   
1          1   both         Soda        77              7     9.0   
2          1   both         Soda        77              7     9.0   
3          1   both         Soda        77              7     9.0   
4          1   both         Soda        77              7     9.0   
5          1   both         Soda        77              7     9.0   
6          1   both         Soda        77              7     9.0   
7          1   both         Soda        77              7     9.0   
8          1   both         Soda        77              7     9.0   
9          1   both         Soda        77              7     9.0   

     price_range_loc     busiest_day         busiest_days    busiest_hour  \
0  Mid-range product  Regularly busy  Regularly busy days  average_orders   
1  Mid-range product  Regularly busy      Least busy days  average_orders   
2  Mid-range product  Regularly busy      Least busy days     most_orders   
3  Mid-range product      Least busy      Least busy days  average_orders   
4  Mid-range product      Least busy      Least busy days     most_orders   
5  Mid-range product  Regularly busy  Regularly busy days  average_orders   
6  Mid-range product  Regularly busy         Busiest days     most_orders   
7  Mid-range product  Regularly busy         Busiest days     most_orders   
8  Mid-range product  Regularly busy         Busiest days     most_orders   
9  Mid-range product      Least busy      Least busy days  average_orders   

   max_order  loyalty_flag  average_price  
0         10  New customer       6.367797  
1         10  New customer       6.367797  
2         10  New customer       6.367797  
3         10  New customer       6.367797  
4         10  New customer       6.367797  
5         10  New customer       6.367797  
6         10  New customer       6.367797  
7         10  New customer       6.367797  
8         10  New customer       6.367797  
9         10  New customer       6.367797

In [123]:
# round the average price to the 2nd decimal place
ords_prods_merged['average_price'] = ords_prods_merged['average_price'].round(2)

In [124]:
# check output
ords_prods_merged.head(50)

order_id  user_id  order_number  orders_day_of_the_week  \
0    2539329        1             1                       2   
1    2398795        1             2                       3   
2     473747        1             3                       3   
3    2254736        1             4                       4   
4     431534        1             5                       4   
5    3367565        1             6                       2   
6     550135        1             7                       1   
7    3108588        1             8                       1   
8    2295261        1             9                       1   
9    2550362        1            10                       4   
10   2968173       15            15                       1   
11   1870022       15            17                       2   
12   1911383       15            18                       2   
13   2715276       15            21                       1   
14    487368       15            22                       1   
15   2293453       19             2                       5   
16   1973799       19             5                       6   
17    532817       19             7                       4   
18   1573906       21            10                       3   
19   1593000       31            10                       3   
20   2231262       31            17                       3   
21   2580647       43             6                       4   
22   2187180       43             9                       4   
23   2497897       52             1                       1   
24   1318871       52             2                       1   
25   1261384       52             3                       1   
26    580568       52             5                       2   
27   2428073       52             7                       2   
28    342306       52             9                       1   
29   1520353       52            10                       2   
30   2542086       52            11                       2   
31    944694       52            12                       1   
32   1268191       52            15                       2   
33   1122089       52            17                       2   
34   1498922       52            18                       2   
35    180919       52            19                       3   
36   2695875       52            21                       5   
37   3205960       67             1                       3   
38    970922       67             2                       1   
39    596585       67             3                       1   
40   2037087       67             4                       2   
41   2474111       67             5                       3   
42   1568481       67             6                       3   
43   2048527       67             8                       2   
44   2389031       67             9                       2   
45   2378562       67            10                       1   
46   1515698       67            11                       1   
47   1928050       67            12                       2   
48   1949274       67            13                       1   
49    551869       67            14                       1   

    order_hour_of_day  days_since_prior_order  new_customer  product_id  \
0                   8                     NaN          True         196   
1                   7                    15.0         False         196   
2                  12                    21.0         False         196   
3                   7                    29.0         False         196   
4                  15                    28.0         False         196   
5                   7                    19.0         False         196   
6                   9                    20.0         False         196   
7                  14                    14.0         False         196   
8                  16                     0.0         False         196   
9                   8                   

### b. Create Spending Flag Variable Based on Each Customer's Spending

Note: If the mean of the prices of products purchased by a user is lower than 10, then flag them as a “Low spender.”  If the mean of the prices of products purchased by a user is higher than or equal to 10, then flag them as a “High spender.”

In [125]:
# create high spender flag
ords_prods_merged.loc[ords_prods_merged['average_price'] < 10, 'spending_flag'] = 'Low spender'
ords_prods_merged.loc[ords_prods_merged['average_price'] >= 10, 'spending_flag'] = 'High spender'

In [126]:
# check output
ords_prods_merged.head()

order_id  user_id  order_number  orders_day_of_the_week  order_hour_of_day  \
0   2539329        1             1                       2                  8   
1   2398795        1             2                       3                  7   
2    473747        1             3                       3                 12   
3   2254736        1             4                       4                  7   
4    431534        1             5                       4                 15   

   days_since_prior_order  new_customer  product_id  add_to_cart_order  \
0                     NaN          True         196                  1   
1                    15.0         False         196                  1   
2                    21.0         False         196                  1   
3                    29.0         False         196                  1   
4                    28.0         False         196                  1   

   reordered _merge product_name  aisle_id  department_id  prices  \
0          0   both         Soda        77              7     9.0   
1          1   both         Soda        77              7     9.0   
2          1   both         Soda        77              7     9.0   
3          1   both         Soda        77              7     9.0   
4          1   both         Soda        77              7     9.0   

     price_range_loc     busiest_day         busiest_days    busiest_hour  \
0  Mid-range product  Regularly busy  Regularly busy days  average_orders   
1  Mid-range product  Regularly busy      Least busy days  average_orders   
2  Mid-range product  Regularly busy      Least busy days     most_orders   
3  Mid-range product      Least busy      Least busy days  average_orders   
4  Mid-range product      Least busy      Least busy days     most_orders   

   max_order  loyalty_flag  average_price spending_flag  
0         10  New customer           6.37   Low spender  
1         10  New customer           6.37   Low spender  
2         10  New customer           6.37   Low spender  
3         10  New customer           6.37   Low spender  
4         10  New customer           6.37   Low spender

In [127]:
# check value counts
ords_prods_merged['spending_flag'].value_counts(dropna = False)

Low spender     31769965
High spender      634894
Name: spending_flag, dtype: int64

## 5. Create Frequency Flag

If the median of “days_since_prior_order” is higher than 20, then the customer should be labeled a “Non-frequent customer.”\
If the median is higher than 10 and lower than or equal to 20, then the customer should be labeled a “Regular customer.”\
If the median is lower than or equal to 10, then the customer should be labeled a “Frequent customer.”

### a. Create Variable for Median Order Frequency

In [128]:
# create median frquency in days since last order
ords_prods_merged['median_ord_freq'] = ords_prods_merged.groupby(['user_id'])['days_since_prior_order'].transform(np.median)

In [129]:
# check output
ords_prods_merged.head()

order_id  user_id  order_number  orders_day_of_the_week  order_hour_of_day  \
0   2539329        1             1                       2                  8   
1   2398795        1             2                       3                  7   
2    473747        1             3                       3                 12   
3   2254736        1             4                       4                  7   
4    431534        1             5                       4                 15   

   days_since_prior_order  new_customer  product_id  add_to_cart_order  \
0                     NaN          True         196                  1   
1                    15.0         False         196                  1   
2                    21.0         False         196                  1   
3                    29.0         False         196                  1   
4                    28.0         False         196                  1   

   reordered _merge product_name  aisle_id  department_id  prices  \
0          0   both         Soda        77              7     9.0   
1          1   both         Soda        77              7     9.0   
2          1   both         Soda        77              7     9.0   
3          1   both         Soda        77              7     9.0   
4          1   both         Soda        77              7     9.0   

     price_range_loc     busiest_day         busiest_days    busiest_hour  \
0  Mid-range product  Regularly busy  Regularly busy days  average_orders   
1  Mid-range product  Regularly busy      Least busy days  average_orders   
2  Mid-range product  Regularly busy      Least busy days     most_orders   
3  Mid-range product      Least busy      Least busy days  average_orders   
4  Mid-range product      Least busy      Least busy days     most_orders   

   max_order  loyalty_flag  average_price spending_flag  median_ord_freq  
0         10  New customer           6.37   Low spender             20.5  
1         10  New customer           6.37   Low spender             20.5  
2         10  New customer           6.37   Low spender             20.5  
3         10  New customer           6.37   Low spender             20.5  
4         10  New customer           6.37   Low spender             20.5

### b. Create Frequency Flag from med_ord_freq

In [130]:
# create frquency flag
ords_prods_merged.loc[ords_prods_merged['median_ord_freq'] > 20, 'Frequency_flag'] = 'Non-frequent customer'
ords_prods_merged.loc[(ords_prods_merged['median_ord_freq'] > 10) & (ords_prods_merged['median_ord_freq'] <= 20), 'Frequency_flag'] = 'Regular customer'
ords_prods_merged.loc[ords_prods_merged['median_ord_freq'] <= 10, 'Frequency_flag'] = 'Frequent customer'

In [131]:
# check output
ords_prods_merged.head()

order_id  user_id  order_number  orders_day_of_the_week  order_hour_of_day  \
0   2539329        1             1                       2                  8   
1   2398795        1             2                       3                  7   
2    473747        1             3                       3                 12   
3   2254736        1             4                       4                  7   
4    431534        1             5                       4                 15   

   days_since_prior_order  new_customer  product_id  add_to_cart_order  \
0                     NaN          True         196                  1   
1                    15.0         False         196                  1   
2                    21.0         False         196                  1   
3                    29.0         False         196                  1   
4                    28.0         False         196                  1   

   reordered _merge product_name  aisle_id  department_id  prices  \
0          0   both         Soda        77              7     9.0   
1          1   both         Soda        77              7     9.0   
2          1   both         Soda        77              7     9.0   
3          1   both         Soda        77              7     9.0   
4          1   both         Soda        77              7     9.0   

     price_range_loc     busiest_day         busiest_days    busiest_hour  \
0  Mid-range product  Regularly busy  Regularly busy days  average_orders   
1  Mid-range product  Regularly busy      Least busy days  average_orders   
2  Mid-range product  Regularly busy      Least busy days     most_orders   
3  Mid-range product      Least busy      Least busy days  average_orders   
4  Mid-range product      Least busy      Least busy days     most_orders   

   max_order  loyalty_flag  average_price spending_flag  median_ord_freq  \
0         10  New customer           6.37   Low spender             20.5   
1         10  New customer           6.37   Low spender             20.5   
2         10  New customer           6.37   Low spender             20.5   
3         10  New customer           6.37   Low spender             20.5   
4         10  New customer           6.37   Low spender             20.5   

          Frequency_flag  
0  Non-frequent customer  
1  Non-frequent customer  
2  Non-frequent customer  
3  Non-frequent customer  
4  Non-frequent customer

## Export Data

In [133]:
# Export to pickle
ords_prods_merged.to_pickle(os.path.join(path_git, '02 Data', 'ords_prods_merged_3.pkl'))